File: src/test/java/org/proshin/blog/oauth/instagram/ErrorTest.java
package org.proshin.blog.oauth.instagram;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;
import org.junit.Test;
import org.proshin.blog.socialnetwork.instagram.oauth.response.Error;

public class ErrorTest {
    @Test
    public void testThatErrorIsParsedFromJson() {
        Error error =
                new Error("" +
                        "{" +
                        "  \"error_type\": \"OAuthException\", " +
                        "  \"code\": 400, " +
                        "  \"error_message\": \"You must provide a client_id\" " +
                        "}");

        assertThat(error.getCode(), is(400));
        assertThat(error.getErrorType(), is("OAuthException"));
        assertThat(error.getErrorMessage(), is("You must provide a client_id"));
    }
}

File: src/test/kotlin/org/proshin/blog/oauth/instagram/ErrorTest.kt
package org.proshin.blog.oauth.instagram

import org.junit.Test
import org.proshin.blog.socialnetwork.instagram.oauth.response.Error
import kotlin.test.expect

class ErrorTest {
    @Test
    fun testThatErrorIsParsedFromJson() {
        val error = Error("""{
                "error_type": "OAuthException",
                "code": 400,
                "error_message": "You must provide a client_id"
            }""")

        expect(400, { error.code })
        expect("OAuthException", { error.errorType })
        expect("You must provide a client_id", { error.errorMessage })
    }
}